### ISSUE TITLE
Voice attack speech functions not working

### ISSUE THREAD

**Issue by HorribleNerd** [EDCD/EDDI#2265]
## What's Wrong (please be as specific as possible)
Calling the "say" or "speech" functions through VoiceAttack causes a NullReferenceException. Using EDDI standalone works just fine, testing the tts works, the speech responder works. Likewise, VoiceAttack also works fine.

### Expected
EDDI is supposed to say the script provided by VoiceAttack.

### Observed
The script gets written to the speechresponder log just fine, but nothing gets said.

### Steps to reproduce
Using VoiceAttack, set the script to some string, then call the EDDI 4.0.1 plugin using context 'say'.

### Configuration
* Version [Please give full version numbers here, not just "latest" or "current"]
  * EDDI Version: 4.0.1
  * VoiceAttack Version: v1.8.9 64-bit, v1.9.9.60 64-bit (beta)
* Localization [if applicable]
  * EDDI's language: en-us
  * Voice: Any voice, no matter the language

### Investigation Notes
I've tried updating both VoiceAttack and EDDI, reinstalling EDDI, changing voices in both EDDI, VoiceAttack and Windows settings. Of note is that it used to work when I last played a year ago, but it stopped working when I updated EDDI for Odyssey.

### EDDI Logs
The logs are perfectly normal, except for the following exception whenever I trigger a speech function:
`2021-10-01T12:54:02 [Warning] SpeechService:getSpeechStream Speech failed (Western European (Windows)): {"ClassName":"System.NullReferenceException","Message":"Object reference not set to an instance of an object.","Data":null,"InnerException":null,"HelpURL":null,"StackTraceString":"   at EddiSpeechService.SpeechService.getSpeechStream(String voice, String speech)","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":"8\ngetSpeechStream\nEddiSpeechService, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null\nEddiSpeechService.SpeechService\nSystem.IO.Stream getSpeechStream(System.String, System.String)","HResult":-2147467261,"Source":"EddiSpeechService","WatsonBuckets":null}`

### Player journals
Not relevant, as it also occurs whenever Elite Dangerous isn't running, and everything else works fine.

**Comment by Tkael** [comment]
My initial attempts to replicate this problem have failed. 

Would you mind please posting screenshots of 
1. your settings in the Text-to-Speech tab of EDDI's UI and
2. a command in VoiceAttack that exhibits this behavior

Would you please also zip and attach `speech.json` from the `%appdata%/EDDI` folder so that we can inspect the raw config file?

**Comment by HorribleNerd** [comment]
EDDI TTS settings:
![image](https://user-images.githubusercontent.com/35268677/135772412-e2d4b464-2ac6-4906-a8aa-83f303d08489.png)

VoiceAttack command:
![image](https://user-images.githubusercontent.com/35268677/135772478-c873356d-96dd-40fc-a04a-e350c2f8a145.png)

Speech.json
[speech.zip](https://github.com/EDCD/EDDI/files/7274567/speech.zip)

**Comment by HorribleNerd** [comment]
My best guess after taking a quick look at the source is that [SpeechService.speak](https://github.com/EDCD/EDDI/blob/1394923781115764122c66391d2a528f973d34e7/SpeechService/SpeechService.cs#L350) is the origin of the null reference. Not sure why it would fail though.

**Comment by Tkael** [comment]
What Windows version are you using?

**Comment by Tkael** [comment]
I agree, either your configuration is somehow null (unlikely from the screenshot and copy of the config file we inspected) or `SpeechService.speak` is returning a null value.

**Comment by HorribleNerd** [comment]
Windows 10 Home version 21H1 (10.0.19043), although it was already happening while I was still on 20H2. It also doesn't appear to be an antivirus conflict either.

**Comment by HorribleNerd** [comment]
Visual Studio debugging confirmed that the exception is thrown at [`if (stream.Length == 0L)`](https://github.com/EDCD/EDDI/blob/1394923781115764122c66391d2a528f973d34e7/SpeechService/SpeechService.cs#L323). `voice` and `speech` get passed to `getSpeechStream` just fine, but for some reason I cannot comprehend `voice` is null inside of `SpeechService.speak`.

**Comment by Tkael** [comment]
Is a voice and synth succesfully selected here?
https://github.com/EDCD/EDDI/blob/1394923781115764122c66391d2a528f973d34e7/SpeechService/SpeechService.cs#L345

Are both synths available here?
https://github.com/EDCD/EDDI/blob/1394923781115764122c66391d2a528f973d34e7/SpeechService/SpeechService.cs#L350-L361

If you are using a voice from the Windows Media synthesizer, is the line below returning null?
https://github.com/EDCD/EDDI/blob/1394923781115764122c66391d2a528f973d34e7/SpeechService/SpeechSynthesizers/WindowsMediaSynthesizer.cs#L180

Verbose logging might also provide a little more context.

**Comment by HorribleNerd** [comment]
When using a Voice Attack speech function:
1. voiceDetails is set to null because voice is null
2. Both synths are skipped because voiceDetails is null
3. Doesn't even get called

When using EDDI standalone:
1. Voice and synthType are set as normal
2. Both are set fine
3. Stream gets returned correctly

Verbose logging just confirms the issue when using VA:
`[Debug] SpeechService:Speak getSpeechStream() returned null; nothing to say`

When using EDDI on its own, verbose logging just displays the speech info (stream length, duration, etc.).

I assume that this is just a really weird issue that is very specific to my machine for some reason.

**Comment by Tkael** [comment]
Ok, I have a theory. This can occur if the user overrides the default voice with a voice name which doesn't match any valid voice.

This succeeds:
![image](https://user-images.githubusercontent.com/4382539/136339979-e0a5d162-56e9-4077-9025-3752ea7e070f.png)

This fails:
![image](https://user-images.githubusercontent.com/4382539/136340304-8c504bcd-9428-4134-b1f3-cc4cdb3280b0.png)

The exception recorded from the failure looks very much like the one you shared:
```
2021-10-07T07:35:07 [Warning] SpeechService:getSpeechStream Speech failed (Western European (Windows)): {"ClassName":"System.NullReferenceException","Message":"Object reference not set to an instance of an object.","Data":null,"InnerException":null,"HelpURL":null,"StackTraceString":"   at EddiSpeechService.SpeechService.getSpeechStream(String voice, String speech) in H:\\Libraries\\source\\repos\\EDDI\\SpeechService\\SpeechService.cs:line 323","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":"8\ngetSpeechStream\nEddiSpeechService, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null\nEddiSpeechService.SpeechService\nSystem.IO.Stream getSpeechStream(System.String, System.String)","HResult":-2147467261,"Source":"EddiSpeechService","WatsonBuckets":null}
```

From a coding perspective, we ought to validate the user input and make sure a valid voice exists before we try to generate a speech stream. We can improve that.

Please check whether you are perhaps setting {TXT:Voice} during some command in your VoiceAttack profile (including from any linked child profiles, searching for any action which sets a text value for that variable name)? If you find such an action then please verify that the voice you're selecting is an *exact* match for the voice name you see in EDDI's UI with no added or missing characters.

**Comment by Tkael** [comment]
Note that if `voice` is null before we start then we select a default voice at lines 302-315. We skip over those lines when `voice` is not null.

**Comment by HorribleNerd** [comment]
That worked! Turns out I had an init profile that set the voice. Disabling the assignment made everything work. Still strange, as I'm quite sure I hadn't changed the init and it used to work fine a year ago. Perhaps an EDDI/Windows update changed the name? (Or I just forgot I had changed it).

In any case, thank you very much! Might I suggest a more specific error message, so other people don't make the same mistake I made? :P
